resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -151,6 +151,13 @@ package object config extends Logging {
       .stringConf
       .createOptional
 
+  private[spark] val KUBERNETES_EXECUTOR_POD_NAME_PREFIX =
+    ConfigBuilder("spark.kubernetes.executor.podNamePrefix")
+      .doc("Prefix to use in front of the executor pod names.")
+      .internal()
+      .stringConf
+      .createWithDefault("spark")
```

Comment:
It's tricky because it's always set on the submission client, but in order to get the value directly we need to use `sparkConf.get(...)`. If the parameter is a config entry of type optional, then we need to call `.get` on the option, and require it to be present. I think it's ok if the contract is that it's always provided but there's still a default value here. At worst the executor names are incorrect but since we're doing all of our logic based on the ID label, it should not matter for correctness.